src/CommunityToolkit.Maui/Converters/IndexToArrayItemConverter.shared.cs

```diff
@@ -17,24 +16,19 @@ public class IndexToArrayItemConverter : ValueConverterExtension, ICommunityTool
 	/// <param name="parameter">The items array.</param>
 	/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
 	/// <returns>The item from the array that corresponds to passed index.</returns>
-	public object? Convert([NotNull] object? value, Type? targetType, [NotNull] object? parameter, CultureInfo? culture)
+	public override object? ConvertFrom(int value, Type? targetType, [NotNull] object? parameter, CultureInfo? culture)
 	{
-		if (value is not int index)
-		{
-			throw new ArgumentException("Value is not a valid integer", nameof(value));
-		}
-
 		if (parameter is not Array array)
 		{
 			throw new ArgumentException("Parameter is not a valid array", nameof(parameter));
 		}
 
-		if (index < 0 || index >= array.Length)
+		if (value < 0 || value >= array.Length)
 		{
 			throw new ArgumentOutOfRangeException(nameof(value), "Index was out of range");
 		}
 
-		return array.GetValue(index);
+		return array.GetValue(value);
```

Comment:
Let's name it index

Reply:
This is just the name of the parameter, `int value`, that is required by `abstract class  BaseConverter<int, object?>`:

```cs
public abstract TTo ConvertFrom(TFrom value, Type targetType, object? parameter, CultureInfo? culture);
```